dev/bots/analyze-sample-code.dart

```diff
@@ -2,6 +2,42 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// This script analyzes all the sample code in API docs in the Flutter source.
+//
+// It uses the following conventions:
+//
+// Code is denoted by markdown ```dart / ``` markers.
+//
+// Only code in "## Sample code" or "### Sample code" sections is examined.
+//
+// There are several kinds of sample code you can specify:
+//
+// * Constructor calls, typically showing what might exist in a build method.
+//   These start with "new" or "const", and will be inserted into an assignment
+//   expression assigning to a variable of type "dynamic" and followed by a
+//   semicolon, for the purposes of analysis.
+//
+// * Class definitions. These start with "class", and are analyzed verbatim.
+//
+// * Other code. It gets included verbatim, though any line that says "// ..."
+//   is considered to separate the block into multiple blocks to be processed
+//   individually.
+//
+// In addition, you can declare code that should be included in the analysis but
+// not shown in the API docs by adding a comment "// Examples can assume:" to
+// the file (usually at the top of the file, after the imports), following by
+// one or more commented-out lines of code. That code is included verbatim in
+// the analysis.
+//
+// All the sample code of every file is analyzed together. This means you can't
+// have two pieces of sample code that define the same example class, for
+// instance.
```

Comment:
uber-nit: I'd just drop the ', for instance' here.

Reply:
Roger. Will fix in subsequent PR.

Reply:
Actually I have to fix merge conflicts anyway, so I'll fix that now.